src/wallet/wallet2.cpp

```diff
@@ -7513,7 +7530,7 @@ std::string wallet2::decrypt(const std::string &ciphertext, const crypto::secret
     THROW_WALLET_EXCEPTION_IF(!crypto::check_signature(hash, pkey, signature),
       error::wallet_internal_error, "Failed to authenticate ciphertext");
   }
-  crypto::chacha8(ciphertext.data() + sizeof(iv), ciphertext.size() - prefix_size, key, iv, &plaintext[0]);
+  crypto::chacha20(ciphertext.data() + sizeof(iv), ciphertext.size() - prefix_size, key, iv, &plaintext[0]);
```

Comment:
Should this handle data encrypted with an older wallet?

Reply:
I did not think this was worth the bother since that's used for export/import files. It's possible to bump the version numbers and do it, though.